test/testpath.cpp

```diff
@@ -247,6 +267,10 @@ private slots:
         QCOMPARE(Path(uR"(c:\)"_s).rootItem(), Path(uR"(c:/)"_s));
         QCOMPARE(Path(uR"(c:\a)"_s).rootItem(), Path(uR"(c:\)"_s));
         QCOMPARE(Path(uR"(c:\a\b)"_s).rootItem(), Path(uR"(c:\)"_s));
+
+        QCOMPARE(Path(uR"(\\nas01\drive)"_s).rootItem(), Path(uR"(\\nas01\drive)"_s));
+        QCOMPARE(Path(uR"(\\nas01\drive\xxx)"_s).rootItem(), Path(uR"(\\nas01\drive)"_s));
+        QCOMPARE(Path(uR"(\\nas01\drive\xxx\yyy)"_s).rootItem(), Path(uR"(\\nas01\drive)"_s));
```

Comment:
Why `\\nas01\drive` is root item?

Reply:
It is the ordinary usecase where we mount a network drive on windows, are there usecases where a UNC path is usable in form of ```\\host``` without specifying the share component?

I don't have a nas at home handy, the test is done by mounting 2 folder as network drive
```
C:\Users\Z>net use
New connections will be remembered.


Status       Local     Remote                    Network

-------------------------------------------------------------------------------
OK           X:        \\mydesktop\mount1
                                                Microsoft Windows Network
OK           Y:        \\mydesktop\mount2
                                                Microsoft Windows Network
The command completed successfully.
```

Reply:
Haven't done any tests with this PR yet, but here's what `net use` returns on my setup (xxx.xxx.xxx.xxx = IP) with 3x NAS connected.

````
C:\WINDOWS\system32>net use
New connections will be remembered.


Status       Local     Remote                    Network

-------------------------------------------------------------------------------
OK           T:        \\xxx.xxx.xxx.xxx\Movies   Microsoft Windows Network
OK           V:        \\xxx.xxx.xxx.xxx\NAS3     Microsoft Windows Network
OK           W:        \\xxx.xxx.xxx.xxx\Tv Shows
                                                Microsoft Windows Network
The command completed successfully.
```

Reply:
> Why `\\nas01\drive` is root item?

Can `\\servername\share` not be used like the docs show?